onnxruntime/core/session/custom_ops.cc

```diff
@@ -826,11 +848,35 @@ ONNX_NAMESPACE::OpSchema CreateSchema(const std::string& domain, const OrtCustom
                     "cannot be inferred without which model loading cannot proceed.");
       }
     }
+
+    std::unordered_set<ONNXTensorElementDataType> all_types;
+    for (auto o : ops) {
+      ORT_ENFORCE(i < o->GetOutputTypeCount(o), "Another version of operator '", schema.Name(), "'has less outputs.");
+      const auto otype = o->GetOutputType(o, i);
+      if (otype == ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED) {
+        all_types.clear();
+        break;
+      }
+      all_types.insert(otype);
+    }
+
```

Comment:
Can we create a helper or lambda for this processing so there's a shared implementation for processing input and outputs to reduce binary size, as it looks pretty similar for both?

Reply:
sure